### PR TITLE
Prevents RCDs from blowing up on central command if ai triggers its ability

### PR DESCRIPTION
**Pull request by 1080pCat**
## What Does This PR Do

Pretty much the title, makes it so the malf ai ability that explodes the RCD can't explode RCDs on central commands z level
## Why It's Good For The Game

image

(station side things shouldn't be effecting cc)

## Testing

Spawned RCDs on lavaland, went ai, went malf mode, blew up rcds, got logs about it exploding lavaland RCDs, and RCDs on station, but not on CC
## Changelog
:cl:
tweak: Malf ai can no longer effect central command with the RCD explode ability
/:cl:

## Notes
I've edited this PR mostly to reflect it not effecting CC

### CODE REVIEW

**Review by DGamerL — APPROVED** [review]
Really just looks fine to me

File: code/game/gamemodes/malfunction/Malf_Modules.dm [hunk 1]
```diff
@@ -376,6 +376,9 @@
 
 /datum/action/innate/ai/destroy_rcds/Activate()
 	for(var/obj/item/rcd/RCD in GLOB.rcd_list)
+		var/turf/RCD_turf = get_turf(RCD)
+		if(RCD_turf.z != usr.z)
```
Comment (hal9000PR) [inline code/game/gamemodes/malfunction/Malf_Modules.dm:380]: might be better to check for station Z level. just my opinion

**Comment by matttheficus** [comment]
This should probably only check for CC's zlevel. You shouldnt have a "safe" RCD by just slipping across zlevels.

**Comment by 1080pCat** [comment]
I've edited this pr to reflect the changes I've made to do with things Matt had concerns about.

File: code/game/gamemodes/malfunction/Malf_Modules.dm (hunk 1 above)
Comment (1080pCat) [reply code/game/gamemodes/malfunction/Malf_Modules.dm]: technicality resolved with latest commit.

**Review by DGamerL — APPROVED** [review]
Didn't quickly find a define for the CC z level, so this LGTM

File: code/game/gamemodes/malfunction/Malf_Modules.dm [hunk 2]
```diff
@@ -376,7 +376,10 @@
 
 /datum/action/innate/ai/destroy_rcds/Activate()
 	for(var/obj/item/rcd/RCD in GLOB.rcd_list)
-		if(!istype(RCD, /obj/item/rcd/borg)) //Ensures that cyborg RCDs are spared.
+		var/turf/RCD_turf = get_turf(RCD)
+		if(istype(RCD, /obj/item/rcd/borg)) //Ensures that cyborg RCDs are spared.
+			continue
+		if(RCD_turf.z != 1)
```
Comment (Contrabang) [inline code/game/gamemodes/malfunction/Malf_Modules.dm:382]: Reachable levels include: The station, lavaland, space levels, but no other levels
```suggestion
		if(is_level_reachable(RCD_turf.z))
```

**Review by Contrabang — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/malfunction/Malf_Modules.dm (hunk 2 above)
Comment (1080pCat) [reply code/game/gamemodes/malfunction/Malf_Modules.dm]: Done

**Review by hal9000PR — APPROVED** [review]

File: code/game/gamemodes/malfunction/Malf_Modules.dm [hunk 3]
```diff
@@ -376,7 +376,10 @@
 
 /datum/action/innate/ai/destroy_rcds/Activate()
 	for(var/obj/item/rcd/RCD in GLOB.rcd_list)
-		if(!istype(RCD, /obj/item/rcd/borg)) //Ensures that cyborg RCDs are spared.
+		var/turf/RCD_turf = get_turf(RCD)
+		if(istype(RCD, /obj/item/rcd/borg)) //Ensures that cyborg RCDs are spared.
+			continue
+		if(is_level_reachable(RCD_turf.z))
 			RCD.detonate_pulse()
```
Comment (hal9000PR) [inline code/game/gamemodes/malfunction/Malf_Modules.dm:383]: get turf after the istype check

**Review by hal9000PR — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/malfunction/Malf_Modules.dm [hunk 4]
```diff
@@ -376,7 +376,10 @@
 
 /datum/action/innate/ai/destroy_rcds/Activate()
 	for(var/obj/item/rcd/RCD in GLOB.rcd_list)
-		if(!istype(RCD, /obj/item/rcd/borg)) //Ensures that cyborg RCDs are spared.
+		var/turf/RCD_turf = get_turf(RCD)
+		if(istype(RCD, /obj/item/rcd/borg)) //Ensures that cyborg RCDs are spared.
+			continue
```
Comment (Contrabang) [inline code/game/gamemodes/malfunction/Malf_Modules.dm:381]:
```suggestion
		if(istype(RCD, /obj/item/rcd/borg)) //Ensures that cyborg RCDs are spared.
			continue
		var/turf/RCD_turf = get_turf(RCD)
```

Comment (1080pCat) [reply code/game/gamemodes/malfunction/Malf_Modules.dm]: done.

File: code/game/gamemodes/malfunction/Malf_Modules.dm (hunk 3 above)
Comment (1080pCat) [reply code/game/gamemodes/malfunction/Malf_Modules.dm]: resolved with contras review

**Review by Contrabang — APPROVED** [review]
LGTM

**Review by Henri215 — APPROVED** [review]

**Review by SteelSlayer — APPROVED** [review]

**Pull request merged**